### PR TITLE
Gin support

### PR DESCRIPTION
**Pull request by tyranron**
From #24 for #22

Implements a `wrapper/gin` subpackage as drop-in replacement for [Gin Gonic](https://gin-gonic.github.io/gin) users.

Example and simple tests are included.

### CODE REVIEW

**Comment by tyranron** [comment]
@rs hmm... should I update Travis CI matrix?

File: cors.go [hunk 1]
```diff
@@ -87,7 +87,9 @@ type Cors struct {
 	// Set to true when allowed headers contains a "*"
 	allowedHeadersAll bool
 	allowCredentials  bool
-	optionPassthrough bool
+	// OptionsPassthrough indicates if preflight lets other potential next
+	// handlers to process the OPTIONS method.
+	OptionPassthrough bool
```
Comment (rs) [inline cors.go]: I would prefer not to expose this. Couldn't we create a wrapping type in the gin wrapper package to keep the knowledge of this option in the gin wrapper side without having to access it from the actual cors.Cors type?

```go
type corsWrapper struct {
    cors.Cors
    optionPassthrough bool
}

func New(options Options) gin.HandlerFunc {
    return corsWrapper{cors.New(options), options.OptionPassthrough}
}

// implement gin.Handler interface on corsWrapper
```

Comment (tyranron) [reply cors.go]: @rs yes, but that won't work if we'll try to wrap and existing `cors.Cors` handler with `Wrap()` (it transforms any `*cors.Cors` to `gin.HandlerFunc`).

Another option I see here is to make a public method:
```go
func (c *Cors) IsPassthroughtEnabled() bool {
    return c.optionPassthrough
}
```

That will disallow `c.optionPassthrough` mutation.

**Comment by rs** [comment]
In what context would you need to wrap an existing cors?

File: cors.go (hunk 1 above)
Comment (tyranron) [reply cors.go]: @rs 

>>>
In what context would you need to wrap an existing cors?
>>>

That may be useful for someone reusing/extending existing code.

However that's not an wide spread case. And even then someone can reuse options, though.

Ok. Then we just can hide `Wrap()` method.

**Comment by tyranron** [comment]
@rs can we switch to go 1.9 in Travis CI?

I'd rather not make `gin.Options` as new type `type Options cors.Options` and leave it a type alias `type Options = cors.Options` for better interop.

**Comment by rs** [comment]
Ok

**Comment by rs** [comment]
I just looked at the gin doc. Why don't you just use `gin.WrapF`?

**Comment by tyranron** [comment]
@rs  because `gin.WrapH()` won't abort Gin middlewares chain for preflights. To abort middlewares chain someone somewhere should call `ctx.Abort()`.

**Comment by rs** [comment]
Ok got it, thx.

**Pull request merged**